http/jsonapi/generator/internal/pay/open-api_test.go

```diff
@@ -135,12 +136,16 @@ CreatePaymentMethodSEPAHandler handles request/response marshaling and validatio
 func CreatePaymentMethodSEPAHandler(service Service, authBackend AuthorizationBackend) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		defer errors.HandleRequest("CreatePaymentMethodSEPAHandler", w, r)
-		// Authentication Handling
-		// OAuth2 Authentication
-		ctx, ok := authBackend.AuthorizeOAuth2(r, w, "pay:payment-methods:create")
+		var ok bool
+		var ctx context.Context // Profile Key Authentication
+		ctx, ok = authBackend.AuthorizeProfileKey(r, &security.NoOpWriter{})
 		if !ok {
-			// No Error Handling needed, this is already done
-			return
+			// OAuth2 Authentication
+			ctx, ok = authBackend.AuthorizeOAuth2(r, &security.NoOpWriter{}, "pay:payment-methods:create")
```

Comment:
I talked with @threez about this and we think it would be best to have another interface so that an authorisation method can tell us whether it can authorise a request. Something similar to:
```go
interface {
	CanAuthorizeRequest(http.Request) bool
}
```
And we then just call the first authorisation method that returns true and let it allow writing to the response directly. It should be an extra interface for backwards compatibility.

Reply:
This creates the problem that authorisation methods "compete" and only the first one that returns true on `CanAuthorizeRequest()` will be checked. So it would be best to configure authorisation in a way that it is clear which authorisation method to choose just by looking at it, for example using different headers, or prefixes, or encodings. But theoretically any custom authorisation method could check with an authorisation server. We don't restrict what means can be done. But we should add some documentation at least.